### DISCUSSION TITLE
When use --interface=wsgi and --lifespan=on there was something wrong

### DISCUSSION THREAD

**Discussion by ryomahan** [Kludex/uvicorn#1631] [Potential Issue]
When we start a WSGI Application use uvicorn with options: `--interface=wsgi --lifespan=on` there will throw an error like:

```shell
Traceback (most recent call last):
  File ".../python3.9/site-packages/uvicorn/lifespan/on.py", line 86, in main
    await app(scope, self.receive, self.send)
  File ".../python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File ".../python3.9/site-packages/uvicorn/middleware/wsgi.py", line 87, in __call__
    assert scope["type"] == "http"
AssertionError
```

but when we only use `--interface=wsgi` the WSGI Application will run successfully, and the interesting thing is lifespan option default value is "auto" and the auto lifespan is equal with on lifespan, they all use `uvicorn.lifespan.on:LifespanOn`.

There is two ways to resolve:
1. raise error when use `--interface=wsgi` with `uvicorn.lifespan.on:LifespanOn`, whether `--lifespan=on` or `--lifespan=auto`
2. just log error when use `--interface=wsgi` with `--lifespan=on` or `--lifespan=auto`

I choose the second way to resolve it and push a PR.

**Comment by ryomahan** [comment]
the PR #1632

**Comment by Kludex** [comment]
How can I reproduce it?

**Reply by ryomahan** [reply]
Just create a simple WSGI Application such as a flask example

```
# save this as app.py
from flask import Flask

app = Flask(__name__)

@app.route("/")
def hello():
    return "Hello, World!"
```

 and run `uvicorn app:app --interface=wsgi --lifespan=on`

**Reply by Kludex** [reply]
Do you run `uvicorn app:app --interface=wsgi`, and works properly?

**Reply by ryomahan** [reply]
yeah, of course.

**Reply by ryomahan** [reply]
because in the `uvicorn.lifespan.on.py:LifespanOn.main` have code like this:

```python
...
            if self.config.lifespan == "auto":
                msg = "ASGI 'lifespan' protocol appears unsupported."
                self.logger.info(msg)
...
``` 

and the lifespan default value is auto, so when we use `uvicorn app:app --interface=wsgi` this code will help us pass error raise.

**Reply by Kludex** [reply]
I don't see a strong case to do something about this. We don't do validation on CLI parameters, yet at least.

**Reply by ryomahan** [reply]
as we know, lifespan is a ASGI concept, so `--interface=wsgi` should not use with `--lifespan=on` or `--lifespan=auto`.

two way to resolve:
1. when we use both together, raise an error
2. when we use both together, loggin an error msg but don't termination

and now, we get yes when we use `--interface=wsgi --lifespan=auto` and get no when we use `--interface=wsgi --lifespan=on`, but as I say they(`--lifespan=auto` and `--lifespan=on`) all use `uvicorn.lifespan.on:LifespanOn`, that means I use the same thing but get different result.

yeah, maybe there isn't  a strong case to do something, I just find it and push it.
